hooks/shared/useGetRequest/useGetRequest.tsx

```diff
@@ -0,0 +1,179 @@
+    refreshToken,
+    dispatch: authDispatch,
+  } = useAuthContext();
+
+  const fetchData = useCallback(async () => {
+    dispatch({ type: "loading" });
+
+    // If a cache exists for this url, return it
+    // If token is required, cache is automatically disabled
+    if (!options?.requiresToken && options?.useCache && cache.current[url!]) {
+      dispatch({ type: "fetched", payload: cache.current[url!] });
+      return;
+    }
+
+    if (!options?.requiresToken) {
+      if (cancelRequest.current) return;
+      const response = await fetch(url!);
+      const json = await response.json();
+
+      if (!response.ok) {
+        if (cancelRequest.current) return;
+        const error: FetchError = new Error(response.statusText);
+        error.status = response.status;
+        error.message = json?.message;
+        dispatch({ type: "error", payload: error as FetchError });
+        return;
+      }
+
+      dispatch({ type: "fetched", payload: json });
+
+      return;
+    }
+
+    try {
+      const response = await fetch(url!, {
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      });
+      if (!response.ok) {
+        const error: FetchError = new Error(response.statusText);
+        const json = await response.json();
+        error.status = response.status;
+        error.message = json?.message;
+        throw error;
+      }
+
+      const data = (await response.json()) as T;
+      cache.current[url!] = data;
+      if (cancelRequest.current) return;
+
+      dispatch({ type: "fetched", payload: data });
+    } catch (error) {
+      if (cancelRequest.current) return;
+
+      if ((error as FetchError)?.status === 401) {
+        try {
+          const REQUEST_ACCESS_TOKEN_URL = `${process.env.NEXT_PUBLIC_BACKEND_URL}/users/api/token/refresh/`;
+
+          const requestNewAccessToken = await fetch(REQUEST_ACCESS_TOKEN_URL, {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+            body: JSON.stringify({
+              refresh: refreshToken,
+            }),
+          });
+
+          const { access, refresh } = await requestNewAccessToken.json();
+          console.log(access);
+
+
+          if (!requestNewAccessToken.ok) throw new Error();
+
+          authDispatch({
+            type: AuthDispatchTypes.LOGIN,
+            payload: {
+              accessToken: access,
+              refreshToken: refresh,
+              remember: !!refresh,
+              user: null,
+            },
+          });
+        } catch (e) {
+          authDispatch({
+            type: AuthDispatchTypes.LOGOUT,
+          });
+        }
+      }
+      dispatch({ type: "error", payload: error as FetchError });
+    }
+  }, [
+    accessToken,
+    refreshToken,
+    authDispatch,
+    url,
+    options?.requiresToken,
+    options?.useCache,
+  ]);
+
+  useEffect(() => {
+    if (!url) return;
+
+    if (!accessToken) {
```

Comment:
Hi Alex, I believe that there was an oversight to this statement. I think this if statement requires one more condition (&& options?.requiresToken) otherwise the data won't be fetched on mount for endpoints that do not require access tokens

Reply:
Oh yeah thanks rashad for the headsup. 😁 If it were to be this way, then no data will be fetch on first mount even if they don't require authorisation. Thanks for the headsup